pkg/health/health_manager.go

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package health
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"path/filepath"
+
+	"github.com/cilium/hive/cell"
+
+	healthApi "github.com/cilium/cilium/api/v1/health/server"
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/datapath/linux/bigtcp"
+	linuxrouting "github.com/cilium/cilium/pkg/datapath/linux/routing"
+	"github.com/cilium/cilium/pkg/datapath/linux/sysctl"
+	datapath "github.com/cilium/cilium/pkg/datapath/types"
+	"github.com/cilium/cilium/pkg/endpoint"
+	endpointcreator "github.com/cilium/cilium/pkg/endpoint/creator"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/health/defaults"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/mtu"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/pidfile"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	controllerInterval    = 60 * time.Second
+	successfulPingTimeout = 3 * time.Minute
+)
+
+// Cell provides the Cilium health infrastructure that is responsible for
+// checking the connectivity between Cilium nodes and Cilium endpoints.
+var Cell = cell.Module(
+	"cilium-health",
+	"Cilium health infrastructure",
+	cell.Provide(newHealthProvider),
+)
+
+type CiliumHealthManager interface {
+	Init(ctx context.Context, routingInfo *linuxrouting.RoutingInfo) error
+	GetStatus() *models.Status
+}
+
+type ciliumHealthManager struct {
+	logger          *slog.Logger
+	healthSpec      *healthApi.Spec
+	sysctl          sysctl.Sysctl
+	loader          datapath.Loader
+	mtuConfig       mtu.MTU
+	bigTCPConfig    *bigtcp.Configuration
+	endpointCreator endpointcreator.EndpointCreator
+	endpointManager endpointmanager.EndpointManager
+	k8sClientSet    k8sClient.Clientset
+
+	ciliumHealth *CiliumHealth
+}
+
+type ciliumHealthParams struct {
+	cell.In
+
+	Logger          *slog.Logger
+	Lifecycle       cell.Lifecycle
+	HealthSpec      *healthApi.Spec
+	Sysctl          sysctl.Sysctl
+	Loader          datapath.Loader
+	MtuConfig       mtu.MTU
+	BigTCPConfig    *bigtcp.Configuration
+	EndpointCreator endpointcreator.EndpointCreator
+	EndpointManager endpointmanager.EndpointManager
+	K8sClientSet    k8sClient.Clientset
+}
+
+func newHealthProvider(params ciliumHealthParams) CiliumHealthManager {
```

Comment:
nit: I think health provider is a term also used by the hive/health stuff, so this may not be the best name.

Reply:
yep agree - this was also the reason why i renamed the provider to manager :smile: It seems i missed this one. Changed.